index.js

```diff
@@ -109,8 +109,15 @@ function Request(url, options, f, retryConfig) {
       return this._reject(err);
     }
 
+    var statusCode = response.statusCode;
+
     // resolve with the full response or just the body
     response = this.fullResponse ? response : body;
+
+    if (this.options.simple && statusCode >= 300) {
```

Comment:
Hum, I think it should be based on the `errorStrategy` instead of `statusCode >= 300`, could you please update your PR ?

Reply:
I'm not entirely sure I properly understand what you mean. As far as I can see there is no such thing as an `errorStrategy`. Did you mean the `retryStrategy`? What would the appropriate behaviour be in this case? 

Correct me if I'm wrong but I don't think we can resort the result of the retry strategy in order to reject the promise. This would imply that non-retryable errors (for example, a 400 bad request) would not be rejected?

Reply:
Yes sorry, I was too quick to answer.

To be more precise : I think it should be a good idea to implement a errorStrategy support, that would be used for promise rejection and callback(err) call as well.